wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3376 @@
+
+    mxq_rc = MXQ_BuildKey(key_buff, &key_buff_len, MXQ_KEYTYPE_ECC,
+                          MXQ_KEYPARAM_EC_P256R1, keycomplen, keylen,
+                          MXQ_KEYUSE_DATASIGNATURE, ALGO_ECDSA_SHA_256,
+                          MXQ_KEYUSE_NONE, ALGO_NONE, (mxq_u1 *)userKey);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_BuildKey() failed");
+        rc = WC_HW_E;
+        goto end_EccSetKey;
+    }
+
+    HOST_LoadDefaultImportKey(sign_key, &sign_key_len, &sign_key_curve,
+                              &sign_key_type);
+
+    rc = HOST_ECDSA_sign(signature, &signature_len, sign_key, key_buff,
+                         key_buff_len, sign_key_curve);
+    if (rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: HOST_ECDSA_sign() failed");
+        goto end_EccSetKey;
+    }
+
+    mxq_rc = MXQ_ImportKey(obj_id, getSignAlgoFromCurve(sign_key_curve),
+                           PUBKEY_IMPORT_OBJID, key_buff, key_buff_len,
+                           signature, signature_len);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_ImportKey() failed");
+        rc = WC_HW_E;
+        goto end_EccSetKey;
+    }
+
+    /* key stored successfully */
+    key->maxq_ctx.key_pending = 0;
+
+end_EccSetKey:
+    wolfSSL_CryptHwMutexUnLock();
+    return rc;
+}
+#endif /* MAXQ_ECC */
+
+void wc_MAXQ10XX_EccFree(ecc_key* key)
+{
+    if (key->maxq_ctx.key_obj_id == 0) {
+        return;
+    }
+
+    int rc = wolfSSL_CryptHwMutexLock();
+    if (rc != 0) {
+        return;
+    }
+
+    mxq_err_t mxq_rc = MXQ_DeleteObject(key->maxq_ctx.key_obj_id);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_DeleteObject() failed");
+        wolfSSL_CryptHwMutexUnLock();
+        return;
+    }
+
+    free_ecc_key_id(key->maxq_ctx.key_obj_id);
+    key->maxq_ctx.key_obj_id = 0;
+    wolfSSL_CryptHwMutexUnLock();
+}
+
+void wc_MAXQ10XX_Sha256Copy(wc_Sha256* sha256)
+{
+    /* during copying, change to soft hash mode for one context */
+    if (sha256->maxq_ctx.hash_running) {
+        sha256->maxq_ctx.hash_running = 0;
+        sha256->maxq_ctx.soft_hash = 1;
+    }
+}
+
+void wc_MAXQ10XX_Sha256Free(wc_Sha256* sha256)
+{
+    /* release the mutex if a hash operation is running on the maxq10xx device
+     */
+    if (sha256->maxq_ctx.hash_running) {
+        sha256->maxq_ctx.hash_running = 0;
+        sha256->maxq_ctx.soft_hash = 1;
+        wolfSSL_CryptHwMutexUnLock();
+    }
+}
+#endif /* WOLFSSL_MAXQ10XX_CRYPTO */
+
+#ifdef WOLF_CRYPTO_CB
+#ifdef MAXQ_SHA256
+static int maxq10xx_hash_update_sha256(const mxq_u1* psrc, mxq_length inlen,
+                                       int running)
+{
+    mxq_err_t mxq_rc;
+
+    if (running == 0) {
+        mxq_rc = MXQ_MD_Init(ALGO_MD_SHA256);
+        if (mxq_rc) {
+            WOLFSSL_ERROR_MSG("MAXQ: MXQ_MD_Init() failed");
+            return WC_HW_E;
+        }
+    }
+
+    mxq_length data_offset = 0;
+    mxq_length data_len;
+
+    while (inlen) {
+        data_len = (inlen < 2000) ? inlen : 2000;
```

Comment:
Make this `2000` value a define. This appears to be the maximum hash chunk size.